starter-code/database.js

```diff
@@ -133,8 +138,10 @@ class Database {
         // productName is the name of the product that we want to buy
         // Think if you may need to implement two queries chained
         // remeber once it's finish to comment callback('Error buying product');
+        this.userFirstName = database.collection('users').find(userFirstName);
+        this.productName = database.collection('products').find(productName);
```

Comment:
Good call. But the exercise only wants you to add the productName, so you don't need to search the product. After that you will need to use $push, search information about it and ask me if you have doubts.